api/src/main/java/org/pmiops/workbench/genomics/GenomicExtractionService.java

```diff
@@ -274,4 +274,19 @@ public GenomicExtractionJob submitGenomicExtractionJob(DbWorkspace workspace, Db
 
     return new GenomicExtractionJob().status(TerraJobStatus.RUNNING);
   }
+
+  public void abortExtract(String wgsCohortExtractionId) throws ApiException {
+    DbWgsExtractCromwellSubmission dbWgsExtractCromwellSubmission =
+        wgsExtractCromwellSubmissionDao.findOne(Long.valueOf(wgsCohortExtractionId));
```

Comment:
There's an access bug here which allows me to abort any job, assuming I have write access to any workspace. Please add a test to cover this  case if possible.

As a pointer on the bug: note that while the provided workspace identifier access is validated, a connection between the workspace and job ID is never verified.